frontend/app-development/features/textEditor/TextEditor.test.tsx

```diff
@@ -24,18 +58,137 @@ const mockLocalStorage = (() => {
 Object.defineProperty(window, 'localStorage', { value: mockLocalStorage });
 
 describe('TextEditor', () => {
-  it('should render the spinner', () => {
+  it('renders the component', async () => {
+    await render();
+
+    expect(screen.getByText(testTextResourceKey)).toBeInTheDocument();
+    expect(screen.getByText(testTextResourceValue)).toBeInTheDocument();
+  });
+
+  it('updates search query when searching text', async () => {
+    await render();
+
+    const search = '1';
+    const searchInput = screen.getByTestId('text-editor-search-default');
+    await act(() => user.type(searchInput, search));
+
+    expect(mockSetSearchParams).toHaveBeenCalledWith({ search });
+  });
+
+  it('adds new text resource when clicking add button', async () => {
+    const upsertTextResources = jest.fn().mockImplementation(() => Promise.resolve());
+
+    await render({ upsertTextResources });
+
+    const addButton = screen.getByRole('button', { name: 'Ny tekst' });
+    await act(() => user.click(addButton));
+
+    expect(upsertTextResources).toBeCalledTimes(2);
+  });
+
+  it('updates text resource when editing text', async () => {
+    const upsertTextResources = jest.fn().mockImplementation(() => Promise.resolve());
+
+    await render({ upsertTextResources });
+
+    const textarea = screen.getByRole('textbox', { name: 'nb translation' });
+    await act(() => user.clear(textarea));
```

Comment:
Do we get the act waring without act within this test? 🤔

Reply:
I tried running the test without wrapping the interactions in act and I got the 'not wrapped in act(...)' warning

Reply:
Then we keep it as it is for now. I see that we are using act many places within our test anyway. Should be a separate task to investigate why we need to do so. Great work and thanks for testing! 😄